index.js

```diff
@@ -316,6 +315,94 @@ const legacyScriptingRunner = (Zap, zcli, input) => {
     return _.template(templateString, options)(values);
   };
 
+  const ensureIsType = (result, type) => {
+    if (type) {
+      if (type.startsWith('array-')) {
+        if (Array.isArray(result)) {
+          return result;
+        } else if (result && typeof result === 'object') {
+          if (type === 'array-wrap') {
+            return [result];
+          } else {
+            // Find the first array in the response
+            for (const k in result) {
+              const value = result[k];
+              if (Array.isArray(value)) {
+                return value;
+              }
+            }
+          }
+        }
+        throw new Error('JSON results array could not be located.');
+      } else if (type.startsWith('object-')) {
+        if (_.isPlainObject(result)) {
+          return result;
+        } else if (
+          Array.isArray(result) &&
+          result.length > 0 &&
+          _.isPlainObject(result[0])
+        ) {
+          // Used by auth test and auth label
+          return result[0];
+        }
+        throw new Error('JSON results object could not be located.');
+      }
+    }
+    return result;
+  };
+
+  const isObject = (object) => {
+    return !Array.isArray(object) && typeof object === 'object';
+  };
+
+  const handleLegacyParams = (data) => {
+    // see handle_legacy_params in the python backend
+    if (!isObject(data)) {
+      return data;
+    }
+    var params = {};
+    for (var i in data) {
+      if (isObject(data[i])) {
+        var param = [];
+        for (var j in data[i]) {
+          param.push(j + ',' + data[i][j]);
+        }
+        params[i] = param.join('|');
+      } else if (Array.isArray(data[i])) {
+        params[i] = data[i].join();
+      }
+    }
+    return params;
+  };
+
+  const textifyList = (data) => {
+    // see textify_list in the python backend
+    if (!Array.isArray(data)){
+      return data;
+    }
+    var out = '';
+    const allObj = data.every(isObject);
+    const sep = allObj ? '\n' : ',';
+
+    for (var i in data) {
```

Comment:
also, more js fun, it's discouraged to use `for...in` on an array. Use `for..of` or `arr.forEach` instead. More info here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#Array_iteration_and_for...in

Reply:
fixed in 749d174f1c5f3e21ef7c0f833cf758a211d277f1 as well